src/go/rpk/pkg/cli/cluster/partitions/move.go

```diff
@@ -53,115 +54,37 @@ func newMovePartitionReplicasCommand(fs afero.Fs, p *config.Params) *cobra.Comma
 			if h, ok := f.Help([]newAssignment{}); ok {
 				out.Exit(h)
 			}
-
 			p, err := p.LoadVirtualProfile(fs)
 			out.MaybeDie(err, "rpk unable to load config: %v", err)
 			config.CheckExitCloudAdmin(p)
 
 			cl, err := adminapi.NewClient(fs, p)
 			out.MaybeDie(err, "unable to initialize admin client: %v", err)
 
-			var (
-				mu                sync.Mutex
-				newAssignmentList []newAssignment
-				wg                sync.WaitGroup
-				brokerReqs        = make(map[int]struct{})
-				knownNodeCore     = make(map[int]int)
-			)
-
-			// Concurrently parse the requested partitions and
-			// find current replica assignments
-			g, egCtx := errgroup.WithContext(cmd.Context())
-			for _, partition := range partitions {
-				partition := partition
-				g.Go(func() error {
-					if len(topics) > 0 { // foo -p 0:1,2,3
-						for _, t := range topics {
-							_, _, part, err := extractNTP(t, partition)
-							out.MaybeDie(err, "failed to extract topic/partition: %s\n", err)
-
-							if nt := strings.Split(t, "/"); len(nt) == 1 {
-								ns = "kafka"
-								topic = nt[0]
-							} else {
-								ns = nt[0]
-								topic = nt[1]
-							}
-
-							current, err := cl.GetPartition(egCtx, ns, topic, part)
-							out.MaybeDie(err, "unable to get partition: %s\n", err)
-
-							newReplicas, err := configureReplicas(partition, current.Replicas)
-							out.MaybeDie(err, "unable to configure new replicas: %v\n", err)
-
-							mu.Lock()
-							newAssignmentList = append(newAssignmentList, newAssignment{
-								Namespace:   ns,
-								Topic:       topic,
-								Partition:   part,
-								OldReplicas: current.Replicas,
-								NewReplicas: newReplicas,
-							})
-							mu.Unlock()
-						}
-					} else { // -p foo/0:1,2,3
-						ns, topic, part, err := extractNTP("", partition)
-						out.MaybeDie(err, "failed to extract topic/partition: %s\n", err)
-
-						current, err := cl.GetPartition(egCtx, ns, topic, part)
-						out.MaybeDie(err, "unable to get partition: %s\n", err)
-
-						newReplicas, err := configureReplicas(partition, current.Replicas)
-						out.MaybeDie(err, "unable to configure new replicas: %v\n", err)
-
-						mu.Lock()
-						newAssignmentList = append(newAssignmentList, newAssignment{
-							Namespace:   ns,
-							Topic:       topic,
-							Partition:   part,
-							OldReplicas: current.Replicas,
-							NewReplicas: newReplicas,
-						})
-						mu.Unlock()
-					}
-					return nil
-				})
-			}
-
-			if err := g.Wait(); err != nil {
-				out.Die("failed to parse the arguments: %v\n", err)
-			}
+			newAssignmentList, coreAssignmentList, err := parseAssignments(cmd.Context(), cl, partitionsFlag, topics)
```

Comment:
In `parseAssignments`, it does check if `updating cores for replicas on new nodes` but it's available in the old (< 24.2.x) API as it's a one shot. That is,

* In < 24.2
  * we should be able to updating cores for replicas on new nodes
  * we should "not" allow changing replication factor as a safe-guard
* In >= 24.x
  * we shouldn't allow either case

Reply:
I understood by @ztlpn comment in #21562 that:

> Assigning a core to a new replica (i.e. replica not present in the current assignment) should be forbidden.

It was for the whole command, but you have a point, the old endpoint is a one-shot endpoint and this should be allowed.

Reply:
Before pushing the change, can @ztlpn confirm that the behavior in the old endpoint is still the same and we can assign a core to a new replica if node_local_core_assignment is off?

Reply:
While I commented above, have been having the same impression as Alexey. To make the behavior consistent, would be better to forbid core assignments for new nodes. 
Thanks.

Reply:
Thanks, both of you, given the above I will mark this conversation resolved and leave the safeguard as it is right now. Thanks again.